source/common/common/utility.cc

```diff
@@ -100,7 +102,7 @@ std::string DateFormatter::fromTime(const SystemTime& time) const {
   const std::chrono::seconds epoch_time_seconds =
       std::chrono::duration_cast<std::chrono::seconds>(epoch_time_ns);
 
-  const auto& item = cached_time.formatted.find(raw_format_string_);
+  const auto& item = cached_time.formatted.find(std::make_pair(raw_format_string_, local_time_));
```

Comment:
This is a performance sensitive feature. It is hard to say that make a pair is free. If the format string length is larger then size of SSO, it will involve an additional heap allocation.

So I think it would be better to use two different maps to cache local time and utc time.